spec-tests/src/main/java/org/candlepin/spec/bootstrap/data/util/CertificateUtil.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2009 - 2022 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.spec.bootstrap.data.util;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.apache.commons.codec.binary.Base64;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.zip.DataFormatException;
+import java.util.zip.Inflater;
+
+public final class CertificateUtil {
+
+    private CertificateUtil() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Extracts the decoded and uncompressed bodies of the entitlement certificates.
+     *
+     * @param jsonPayload - the json that contains the encoded and compressed entitlement certificates
+     * @return the entitlement certificate bodies
+     * @throws DataFormatException if unable to decompress the body of the certificate
+     * @throws JsonMappingException if unable to parse the json
+     * @throws JsonProcessingException if unable to parse the json
+     */
+    public static List<JsonNode> extractEntitlementCertificatesFromPayload(Object jsonPayload)
```

Comment:
We could theoretically add those DTOs to the spec file. It is part of our API anyway, so we could generate them and link it in a comment as the compressed output.

If not then yes. client.model seems like a good fit.

@nikosmoum @Ceiu @wottop What do you think?

Reply:
I think using DTOs (either from spec file, or manually crafted), would add an abstraction layer that would make our ability to assert on the cert payload properly hard, or even impossible in some cases (this is the problem that @Ceiu has pointed out before with the general DTOs in responses). I think we should keep using JsonNodes for parsing certificate payloads

Reply:
@nikosmoum Using DTOs wouldn't prevent us from using json. We could use DTO version for the general case and still be able to fallback on raw json in those rare cases where we need to. The inverse of this is probably what we will end up doing for getting json instead of DTO.

Reply:
Y'all have my core thoughts in mind already, so I'll skip that. So, that aside, especially in light of the cert unification and stuff I'd like to see, I would really like to avoid making the nested objects a critical or important aspect of the certs if we can.

Reply:
Maybe the best approach is partial DTOs? meaning, instead of adding the whole nested hierarchy, just add the fields we need to assert on (products, content) and not the rest (order, subscription, etc.)?

Also, we'll need a naming convention here. Maybe CertProductDTO/CertContentDTO/etc to differentiate between the other DTOs.

Reply:
I think that can work, but would these classes only be used for spec-tests and would that make it a DTO? Like you said before we do lose that ability to assert on the payload properly as well. I think we should leave it as a json and have a broader discussion about the DTOs in general and what we should do in cases where asserting against the json is necessary.

Reply:
Additionally It just feels like overkill defining DTOs just for the purpose of using them for asserting in spec tests, when the assertions are very simple/exactly the same across all tests (going 2-3 levels in the tree to check the products and content fields)

Reply:
For the sake of not blocking this PR, let's merge now, and discuss the need for defining DTOs elsewhere